synapse/handlers/room_list.py

```diff
@@ -295,19 +311,28 @@ def _append_room_entry_to_chunk(self, room_id, num_joined_users, chunk, limit,
                                     search_filter, from_federation=False):
         """Generate the entry for a room in the public room list and append it
         to the `chunk` if it matches the search filter
+
+        Args:
+            room_id (str): The ID of the room.
```

Comment:
Again, if you're going to add these, please get the types right

Reply:
https://github.com/matrix-org/synapse/blob/40c2271680501c5ca2c29349d62df42d80b953db/synapse/storage/state.py#L486

Are they supposed to be `RoomID`s?

Reply:
Ah, you meant adding `|None`. Have done so for the other types yes.